src/eva/data/data_collections.py

```diff
@@ -188,31 +190,47 @@ def get_variable_data_array(self, collection_name, group_name, variable_name, ch
         """
 
         group_variable_name = group_name + '::' + variable_name
-
         data_array = self._collections[collection_name][group_variable_name]
 
-        if channels is None:
+        if channels is None and levels is None:
             return data_array
-        elif isinstance(channels, int) or not any(not isinstance(c, int) for c in channels):
-            # Channel must be a dimension if it will be used for selection
-            if 'Channel' not in list(self._collections[collection_name].dims):
-                self.logger.abort(f'In get_variable_data_array channels is provided but ' +
-                                  f'Channel is not a dimension in Dataset')
-            # Make sure it is a list
-            channels_sel = []
-            channels_sel.append(channels)
-
-            # Create a new DataArray with the requested channels
-            data_array_channels = data_array.sel(Channel=channels_sel)
-            return data_array_channels
+
+        if channels is not None:
+            if isinstance(channels, int) or not any(not isinstance(c, int) for c in channels):
+                # Channel must be a dimension if it will be used for selection
+                if 'Channel' not in list(self._collections[collection_name].dims):
+                    self.logger.abort(f'In get_variable_data_array channels is provided but ' +
+                                      f'Channel is not a dimension in Dataset')
+                # Make sure it is a list
+                channels_sel = []
+                channels_sel.append(channels)
+
+                # Create a new DataArray with the requested channels
+                data_array_channels = data_array.sel(Channel=channels_sel)
+                return data_array_channels
+
+        elif levels is not None:
+            if isinstance(levels, int) or not any(not isinstance(lev, int) for lev in levels):
+                # Level must be a dimension if it will be used for selection
+                if 'Level' not in list(self._collections[collection_name].dims):
+                    self.logger.abort(f'In get_variable_data_array levels is provided but ' +
+                                      f'Level is not a dimension in Dataset')
+                # Make sure it is a list
+                levels_sel = []
+                levels_sel.append(levels)
+
+                # Create a new DataArray with the requested channels
+                data_array_levels = data_array.sel(Level=levels_sel)
+                return data_array_levels
 
         else:
```

Comment:
I think this else needs modified now, right? Or at very least the error message that gets written out

Reply:
Fair point.  See if these changes handle things better.

Reply:
Yep thanks!